Common/API/LoginHelper.swift

```diff
@@ -19,16 +19,28 @@ public extension UserDefaults {
 
 public class LoginHelper {
 
-    public static func getAccessToken(username: String?, password: String?) -> Future<String, SCError> {
+
+    /// Setup the sync engine with the callback needed when an authentication error happens
+    public static func setupAuthentication(authenticationHandler: @escaping () -> Void) {
+        SyncHelper.authenticationChallengerHandler = authenticationHandler
+    }
```

Comment:
Why not set the handler directly on `SyncEngine`? Is there a reason for this indirection?

Reply:
It does feel more correct that the UI is letting the LoginManager on how handling re-authentication, plus there I might add more logic on this function later for more fine grained handling.